ignite/metrics/loss.py

```diff
@@ -42,40 +42,40 @@ class Loss(Metric):
         and the Loss metric using an ``evaluator`` created with
         :meth:`~ignite.engine.create_supervised_evaluator` method.
 
-        .. code-block:: python
+        .. testcode::
 
-            import torch
-            import torch.nn as nn
-            from torch.nn.functional import nll_loss
-
-            from ignite.metrics import Accuracy, Loss
-            from ignite.engine import create_supervised_evaluator
-
-            model = ...
+            model = nn.Linear(10, 3)
 
             criterion = nll_loss
+            c_kwargs = {"reduction": "sum"}
 
-            metrics = {
-                "Accuracy": Accuracy(),
-                "Loss": Loss(criterion)
-            }
+            def process_function(engine, batch):
```

Comment:
A default trainer is defined in this [file](https://github.com/pytorch/ignite/blob/master/docs/source/conf.py). You must use it. However, a specific setup is needed, please see the following example 

https://github.com/pytorch/ignite/blob/5455fa5885225f7d5e39aab31a796b7efc067768/ignite/handlers/param_scheduler.py#L364

Reply:
i understand how to setup default trainer! 
But i can't get how/where to use default trainer in loss metrics

Reply:
> 

can you give some advice on the question above?

Reply:
Sorry I missed your question !!

This trainer should replace your engine. No longer need to have a process function too. Look the example I mentioned, there is no engine.